scripts/update-merkle-root.ts

```diff
@@ -0,0 +1,180 @@
+}
+
+enum Verification {
+  email = 'email',
+  twitter = 'twitter',
+  AlumNFT = 'AlumNFT',
+  BWLNFT = 'BWLNFT',
+  token = 'token',
+}
+
+enum VerificationType {
+  email = '0',
+  twitter = '1',
+  balance = '2',
+  token = '3',
+}
+
+function generateHashByRecord(
+  str: string,
+  hashFunc: (message: (number | string | BigNumber)[] | Uint8Array) => string
+) {
+  const [verification, content] = str.split(':')
+  switch (verification) {
+    case Verification.token:
+      if (isNaN(parseInt(content, 10)))
+        throw new Error(`Invalid number: ${content}!`)
+      return hashFunc([hexlifyString(content)])
+    case Verification.email:
+      return hashFunc([VerificationType.email, hexlifyString(content)])
+    case Verification.twitter:
+      if (isNaN(parseInt(content, 10)))
+        throw new Error(`Invalid number: ${content}!`)
+      return hashFunc([VerificationType.twitter, parseInt(content, 10)])
+    case Verification.AlumNFT:
+      if (!ethereumAddressRegex.test(content.toLowerCase()))
+        throw new Error(`Invalid ethereum address: ${content}!`)
+      return hashFunc([
+        VerificationType.balance,
+        hexlifyString(content.toLowerCase()),
+        1,
+        hexlifyString(YC_ALUM_NFT_CONTRACT),
+      ])
+    case Verification.BWLNFT:
+      if (!ethereumAddressRegex.test(content.toLowerCase()))
+        throw new Error(`Invalid ethereum address: ${content}!`)
+      return hashFunc([
+        VerificationType.balance,
+        hexlifyString(content.toLowerCase()),
+        1,
+        hexlifyString(KETL_BWL_NFT_CONTRACT),
+      ])
+    default:
+      throw new Error(`Unknow verification type: ${verification}, str: ${str}!`)
+  }
+}
+
+async function main() {
+  const [deployer] = await ethers.getSigners()
+
+  // Deploy the contract
+  console.log('Updating contracts with the account:', deployer.address)
+  console.log(
+    'Account balance:',
+    utils.formatEther(await deployer.getBalance())
+  )
+
+  const promptEnv = await prompt.get({
+    properties: {
+      isProduction: {
+        type: 'boolean',
+        required: true,
+        default: false,
+      },
+    },
+  })
+
+  const promptResult = await prompt.get({
+    properties: {
+      attestationAddress: {
+        required: true,
+        default: promptEnv.isProduction
+          ? PROD_KETL_ATTESTATION_CONTRACT
+          : DEV_KETL_ATTESTATION_CONTRACT,
+        pattern: ethereumAddressRegex,
+      },
+    },
+  })
+
+  const { attestationAddress } = promptResult
+
+  console.log(`Updating contract: ${attestationAddress}`)
+  const ketlAttestation = KetlAttestation__factory.connect(
+    attestationAddress,
+    deployer
+  )
+
+  const hashFunc = await poseidonHash()
+  const ids = [0, 1, 2, 3]
+
+  for (const id of ids) {
+    const filePath = resolve(cwd(), 'merkleTrees', `${id}.txt`)
+    const liner = new lineByLine(filePath)
+
+    let line
+    const attestationHashes = [] as string[]
+    while ((line = liner.next())) {
```

Comment:
Why don't we do `while (const line = liner.next())`?

Reply:
Nope, it doesn't work, sorry